nireports/reportlets/xca.py

```diff
@@ -237,6 +239,11 @@ def plot_melodic_components(
     )
     fig.clf()
 
+    if compress:
+        image_svg = svg_compress(Path(out_file).read_text(), compress=compress)
+        with open(out_file, "w", encoding="utf-8") as f:
+            f.write(image_svg)
```

Comment:
since we're already using pathlib
```suggestion
        Path(out_file).write_text(image_svg)
```